### PR TITLE
Mayhem Heroes Integration

### PR DESCRIPTION
**Pull request by capuanob**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+FROM --platform=linux/amd64 ubuntu:20.04 as builder
+
+## Install build dependencies.
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -y git clang gcc ruby-dev make
+
+## Add source code to the build stage.
+WORKDIR /
+ADD https://api.github.com/repos/capuanob/blurhash/git/refs/heads/mayhem version.json
+RUN git clone -b mayhem https://github.com/capuanob/blurhash.git
```
Comment (xansec) [inline Dockerfile:11]: Please do not use git clone.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+WORKDIR ext/blurhash
+RUN ruby extconf.rb
+RUN make
+RUN cp encode.so /usr/lib
+WORKDIR /blurhash
+RUN clang -fsanitize=fuzzer fuzz/fuzz.c -Lencode -I ext/blurhash/ -L/usr/lib -l:encode.so -o blurhash-fuzz
+
+## Consolidate all dynamic libraries used by the fuzzer
+RUN mkdir /deps
+RUN cp `ldd blurhash-fuzz | grep so | sed -e '/^[^\t]/ d' | sed -e 's/\t//' | sed -e 's/.*=..//' | sed -e 's/ (0.*)//' | sort | uniq` /deps 2>/dev/null || :
```
Comment (xansec) [inline Dockerfile:24]: Please do not use `ldd`.

File: Mayhemfile [hunk 3]
```diff
@@ -0,0 +1,10 @@
+image: ghcr.io/capuanob/blurhash-fuzz:latest
```
Comment (xansec) [inline Mayhemfile:1]: You may omit image name.

File: Mayhemfile [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+project: blurhash
+target: encode-fuzz
+tests:
```
Comment (xansec) [inline Mayhemfile:4]: Tests -> testsuite.

File: Mayhemfile [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+  - file://corpus
+
+cmds:
+  - cmd: /blurhash-fuzz -close_fd_mask=2
+    libfuzzer: true
+    max_length: 300000
```
Comment (xansec) [inline Mayhemfile:10]: Why such a large max_length?

File: fuzz/fuzz.c [hunk 6]
```diff
@@ -0,0 +1,28 @@
+#include <stdint.h> // uint8_t
```
Comment (xansec) [inline fuzz/fuzz.c:1]: Can we rename to indicate what's being fuzzed?

**Review by xansec — CHANGES_REQUESTED** [review]
Please do not upload such a massive corpus. 1-2 seeds should suffice.

**Comment by capuanob** [comment]
@xansec I've made these requested changes, but can't get a high test case count. Any recommendations?

**Comment by xansec** [comment]
We can probably copy the corpus from one of your previous runs to the mayhem heroes fork. LGTM!

**Pull request merged**